_posts/2023-10-06-code-coverage.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Code Coverage for Unit Tests"
+excerpt: "Code coverage helps in the easy maintenance of the codebase, exposure of bad code, and results in faster time to market."
+tags: QA Mobile Android iOS Tests
+authors:
+- Rafael Nascimento
+header:
+  teaser: /assets/images/post/code-coverage/code_coverage.png
+  teaser_alt: Code coverage
+category: QA
+---
+
+![](/assets/images/post/code-coverage/code_coverage.png)
+
+# Introduction
+
+Code coverage is the percentage of lines of your code that is being executed by your tests. This is an indicative of how safe your code is and how bug proof it can be while refactoring or adding new features. Code coverage helps in the easy maintenance of the codebase, exposure of bad code, and results in faster time to market. 
+
+This metric is measured using the following formula:
+
+**_Coverage Percentage = (number of unique lines of code executed / total number of lines of code) * 100._**
```

Comment:
Hi - is this always true? For example, sonar calculates the code coverage with a different formula

```
Coverage = (CT + CF + LC)/(2*B + EL)

where:
* CT = conditions that have been evaluated to 'true' at least once
* CF = conditions that have been evaluated to 'false' at least once
* LC = covered lines = linestocover - uncovered_lines
* B = total number of conditions
* EL = total number of executable lines (lines_to_cover)
```

Reply:
I've revisited this and mentioned both